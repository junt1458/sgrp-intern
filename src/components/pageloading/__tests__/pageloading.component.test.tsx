import { render } from '@testing-library/react';
import PageLoading from '../pageloading.component';

describe('should render the component', () => {
  it('w/ loading', () => {
    const { container } = render(
      <PageLoading isLoading={true} isPermissionError={false}>
        <div>aaaa</div>
      </PageLoading>
    );
    expect(container).toMatchSnapshot();
  });

  it('w/ error', () => {
    const { container } = render(
      <PageLoading isLoading={false} isPermissionError={true}>
        <div>aaaa</div>
      </PageLoading>
    );
    expect(container).toMatchSnapshot();
  });

  it('w/ view', () => {
    const { container } = render(
      <PageLoading isLoading={false} isPermissionError={false}>
        <div>aaaa</div>
      </PageLoading>
    );
    expect(container).toMatchSnapshot();
  });
});
