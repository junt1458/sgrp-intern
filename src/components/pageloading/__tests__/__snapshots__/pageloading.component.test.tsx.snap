// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component w/ error 1`] = `
<div>
  <div
    class="flex h-14 w-full items-center justify-between bg-slate-900 px-4 py-2 text-xl"
  >
    <a
      href="/"
    >
      Internship Application System
    </a>
  </div>
  <div
    class="my-8"
  >
    <div
      class="flex flex-wrap items-center justify-center"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-red-700 css-i4bv87-MuiSvgIcon-root"
        data-testid="ErrorIcon"
        focusable="false"
        style="width: 128px; height: 128px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
        />
      </svg>
      <div
        class="w-full text-center"
      >
        You do not have permission to view this page.
      </div>
    </div>
  </div>
  <div
    class="my-2 flex justify-center"
  >
    <button
      class="rounded-lg px-4 py-2"
      style="background-color: rgb(97, 158, 255); color: rgb(0, 0, 0);"
    >
      Back to home
    </button>
  </div>
</div>
`;

exports[`should render the component w/ loading 1`] = `
<div>
  <div
    class="flex h-14 w-full items-center justify-between bg-slate-900 px-4 py-2 text-xl"
  >
    <a
      href="/"
    >
      Internship Application System
    </a>
  </div>
  <div
    class="my-8"
  >
    <div
      class="flex flex-wrap items-center justify-center"
    >
      <img
        alt="Loading icon"
        height="200"
        src="/loading.gif"
        width="200"
      />
      <div
        class="w-full text-center"
      >
        Loading...
      </div>
    </div>
  </div>
</div>
`;

exports[`should render the component w/ view 1`] = `
<div>
  <div>
    aaaa
  </div>
</div>
`;
